sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2513,4 +2513,18 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test(
+    "SPARK-19059: Unable to retrieve data from parquet table whose name startswith underscore") {
+    sql("CREATE TABLE `_tbl`(i INT) USING parquet")
+    sql("INSERT INTO `_tbl` VALUES (1), (2), (3)")
+    checkAnswer( sql("SELECT * FROM `_tbl`"), Row(1) :: Row(2) :: Row(3) :: Nil)
```

Comment:
I think we can stop here, create a table with underscore, then insert into it, then read it, that's enough to prove we can support table with underscore.